cmd/restic/cmd_diff.go

```diff
@@ -348,7 +378,57 @@ func runDiff(opts DiffOptions, gopts GlobalOptions, args []string) error {
 		return err
 	}
 
-	Verbosef("comparing snapshot %v to %v:\n\n", sn1.ID().Str(), sn2.ID().Str())
+	if !gopts.JSON {
+		Verbosef("comparing snapshot %v to %v:\n\n", sn1.ID().Str(), sn2.ID().Str())
+	}
+
+	var (
+		printChange    func(path string, mode string)
+		printDiffStats func(stats *DiffStats)
+	)
+
+	if gopts.JSON {
+		enc := json.NewEncoder(gopts.stdout)
+
+		printChange = func(path string, mode string) {
+			err := printChangeJSON(enc, path, mode)
+			if err != nil {
+				Warnf("JSON encode failed: %v\n", err)
+			}
+		}
+
+		printDiffStats = func(stats *DiffStats) {
+
+			diff := Diff{
+				ID1:      sn1.ID(),
+				ShortID1: sn1.ID().Str(),
+				ID2:      sn2.ID(),
+				ShortID2: sn2.ID().Str(),
+
+				Stats: *stats,
+			}
+
+			err := printStatsJSON(enc, &diff)
+			if err != nil {
+				Warnf("JSON encode failed: %v\n", err)
+			}
+		}
+	} else {
+		printChange = func(path string, mode string) {
+			Printf("%-5s%v\n", mode, path)
+		}
+
+		printDiffStats = func(stats *DiffStats) {
```

Comment:
Please inline the `printDiffStats` methods and just add an if at the end of `runDiff`.